packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -648,14 +655,13 @@ export class EditorPage {
 
 		const json = await response.json();
 
+		// AT and Simple sites have slightly differing response from the API.
 		let publishedURL: string;
-		if ( this.target === 'atomic' ) {
+		if ( json.link ) {
 			publishedURL = json.link;
-		} else {
+		} else if ( json.body.link ) {
```

Comment:
**suggestion (non-blocking):** The usage of if-else is easier to read, but what are your thoughts if we switch this to a ternary operator instead?

Reply:
I had thought about this a little and the main reason is that the structure of the response differs between AT and Simple, so a ternary may try to access a non-existent attribute and thus throw an error instead, at least from my experiments in the interpreter:

```typescript
> const a = {
... a: 'lkwjerlwer',
... b: {
... c: ''
... }
... }
undefined
> a
{ a: 'lkwjerlwer', b: { c: '' } }
> a.c.d ? true : false
Uncaught TypeError: Cannot read properties of undefined (reading 'd')
```

In this case, we're working with three possible outcomes and while ternaries can be stacked I do not feel the tradeoff is worth the added complexity of deciphering the stack.